src/vse_sync_pp/analyzers/analyzer.py

```diff
@@ -265,3 +270,177 @@ def explain(self, data):
+        self._transient = config.parameter('transient-period/s')
+        # minimum test duration for a valid test
+        self._duration = config.parameter('min-test-duration/s')
+        # limit to 10k taus for now, covers most masks
+        taus_a = np.array([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 20, 30, 40, 50, 60, 70, 80, 90,
+                           100, 200, 300, 400, 500, 600, 700, 800, 900, 1000, 2000, 3000,
+                           4000, 5000, 6000, 7000, 8000, 9000])
+        taus_b = np.arange(10000, 100001, 5000)
+        # predefine limit observation window intervals for which to compute the statistic
+        self._taus_list = np.concatenate((taus_a, taus_b))
+        # tau-list values for which the statistic will be actually computed
+        self._taus = None
+        self._samples = None
+        self._errs = None
+        self._ns = None
+        self._rate = None
+        self._lpf_signal = None
+
+    def prepare(self, rows):
+        idx = 0
+        try:
+            tstart = rows[0].timestamp + self._transient
+        except IndexError:
+            pass
+        else:
+            while idx < len(rows):
+                if tstart <= rows[idx].timestamp:
+                    break
+                idx += 1
+        return super().prepare(rows[idx:])
+
+    def calculate_limit(self, tau):
+        # return limit based on `tau`
+        for (low, high), f in self._accuracy.items():
+            if ((low is None or tau > low) and (tau <= high)):
+                return f(tau) * (self._limit / 100)
+
+    def out_of_range(self, taus, samples):
+        # return true if any value in `tdevs` is out of range
+        for tau, sample in zip(taus, samples):
+            mask = self.calculate_limit(tau)
+            if mask <= sample:
+                return (True)
+        return False
+
+    def calculate_filter(self, input_signal):
+        # returns low-pass filter from `input_signal`
+        # scipy_signal.butter args
+        # order of the filter: 1
+        # cutoff Frequency: 0.1Hz
+        # the critical frequency for digital filters is normalized from 0 to 1 where 1 is Nyquist freqency.
+        # btype is band type or type of filter
+        # analog=False since it is always a digital filter
+        # `numerator` coefficient vector and `denominator coefficient vector of the butterworth digital filter
+        numerator, denominator = scipy_signal.butter(1, 0.1 / (1 / 2), btype="low", analog=False, output="ba")
+        lpf_signal = scipy_signal.filtfilt(numerator, denominator, input_signal.terror)
+        lpf_signal = lpf_signal[self._transient:len(lpf_signal)]
+        return lpf_signal
+
+    def calculate_rate(self, data):
+        # calculate sample rate using 100 samples
+        cumdelta = 0
+        for i in range(1, 100):
+            cumdelta = cumdelta + data.iloc[i].timestamp - data.iloc[i - 1].timestamp
+        return round((1 / (cumdelta / 100)))
+
+    def test(self, data):
```

Comment:
Recommend changing the name of this base method as it does not implement `test` contract. (Rather it is a partial test, assessing common error and failures.)